src/snowflake/snowpark/mock/_plan.py

```diff
@@ -411,26 +414,48 @@ def handle_udf_expression(
     if udf_name not in udf_registry:
         raise SnowparkSQLException(f"[Local Testing] udf {udf_name} does not exist.")
 
+    # Initialize import directory
```

Comment:
If a udf is called multiple times in the same query will the modules be copied for each invocation?

Reply:
Yes, we could potentially optimize by initializing the import directory at UDF registration.